testbook/client.py

```diff
@@ -146,16 +191,31 @@ def inject(self, code, args=None, run=False, before=None, after=None):
 
         return TestbookNode(self.execute_cell(inject_idx)) if run else TestbookNode(code_cell)
 
-    def value(self, name):
-        """Extract a JSON-able variable value from notebook kernel"""
-
-        result = self.inject(name, run=True)
-        if not self._execute_result(result.outputs):
-            raise TestbookError('code provided does not produce execute_result')
+    def value(self, code):
+        result = self.inject(code)
 
-        code = """
-        from IPython.display import JSON
-        JSON({"value" : _})
-        """
-        cell = self.inject(code, run=True)
-        return cell.outputs[0].data['application/json']['value']
+        if self._execute_result(result):
+            code = """
+                from IPython.display import JSON
+                JSON({"value" : _})
+            """
+            cell = self.inject(code)
+            return cell.outputs[0].data['application/json']['value']
+        else:
+            return self._output_text(result)
+
+    @contextmanager
+    def exception_mapper(self):
+        try:
+            yield
+        except CellExecutionError as e:
+            raise eval(e.ename)(e) from None
```

Comment:
This is the output of `inspect.getmembers(e)`:
<details><summary>Output</summary>

```
[('__cause__', None),
 ('__class__', <class 'nbclient.exceptions.CellExecutionError'>),
 ('__context__', None),
 ('__delattr__',  <method-wrapper '__delattr__' of CellExecutionError object at 0x10d426a00>),
 ('__dict__',
  {'ename': 'NameError',
   'evalue': "name 'ref' is not defined",
   'traceback': '...'}),
 ('__dir__',  <built-in method __dir__ of CellExecutionError object at 0x10d426a00>),
 ('__doc__', '...'),
 ('__eq__',  <method-wrapper '__eq__' of CellExecutionError object at 0x10d426a00>),
 ('__format__',  <built-in method __format__ of CellExecutionError object at 0x10d426a00>),
 ('__ge__',  <method-wrapper '__ge__' of CellExecutionError object at 0x10d426a00>),
 ('__getattribute__',  <method-wrapper '__getattribute__' of CellExecutionError object at 0x10d426a00>),
 ('__gt__',  <method-wrapper '__gt__' of CellExecutionError object at 0x10d426a00>),
 ('__hash__',  <method-wrapper '__hash__' of CellExecutionError object at 0x10d426a00>),
 ('__init__',  <bound method CellExecutionError.__init__ of CellExecutionError("...")>),
 ('__init_subclass__',
  <built-in method __init_subclass__ of type object at 0x7ffa68825180>),
 ('__le__',  <method-wrapper '__le__' of CellExecutionError object at 0x10d426a00>),
 ('__lt__',  <method-wrapper '__lt__' of CellExecutionError object at 0x10d426a00>),
 ('__module__', 'nbclient.exceptions'),
 ('__ne__',  <method-wrapper '__ne__' of CellExecutionError object at 0x10d426a00>),
 ('__new__', <built-in method __new__ of type object at 0x10a864a68>),
 ('__reduce__',  <built-in method __reduce__ of CellExecutionError object at 0x10d426a00>),
 ('__reduce_ex__',  <built-in method __reduce_ex__ of CellExecutionError object at 0x10d426a00>),
 ('__repr__',  <method-wrapper '__repr__' of CellExecutionError object at 0x10d426a00>),
 ('__setattr__',  <method-wrapper '__setattr__' of CellExecutionError object at 0x10d426a00>),
 ('__setstate__',  <built-in method __setstate__ of CellExecutionError object at 0x10d426a00>),
 ('__sizeof__',  <built-in method __sizeof__ of CellExecutionError object at 0x10d426a00>),
 ('__str__',
  <bound method CellExecutionError.__str__ of CellExecutionError("...")>),
 ('__subclasshook__',  <built-in method __subclasshook__ of type object at 0x7ffa68825180>),
 ('__suppress_context__', False),
 ('__traceback__', <traceback object at 0x10d43f940>),
 ('__unicode__',  <bound method CellExecutionError.__unicode__ of CellExecutionError("..."),
 ('__weakref__', None),
 ('args', "...",
 ('ename', 'NameError'),
 ('evalue', "name 'ref' is not defined"),
 ('from_cell_and_msg', <bound method CellExecutionError.from_cell_and_msg of <class 'nbclient.exceptions.CellExecutionError'>>),
 ('traceback', '....'),
 ('with_traceback', <built-in method with_traceback of CellExecutionError object at 0x10d426a00>)]
```
</details>


Looks like we would need to make some changes to the way nbclient is raising the error since there is no `__bases__` attribute.